src/EchoParameterDynamicFunctionReturnTypeExtension.php

```diff
@@ -78,13 +78,44 @@ public function getTypeFromFunctionCall(FunctionReflection $functionReflection,
 
         if ($echoArgumentType instanceof ConstantBooleanType) {
             return ($echoArgumentType->getValue() === false)
-                ? new StringType()
-                : new VoidType();
+                ? self::getEchoFalseReturnType($name)
+                : self::getEchoTrueReturnType($name);
         }
 
         return TypeCombinator::union(
-            new StringType(),
-            new VoidType()
+            self::getEchoFalseReturnType($name),
+            self::getEchoTrueReturnType($name)
         );
     }
+
+    protected static function getEchoTrueReturnType( string $name ): Type
+    {
+        if ($name === 'single_month_title') {
+            return TypeCombinator::union(
+                new VoidType(),
+                new ConstantBooleanType(false)
+            );
+        }
+
+        return new VoidType();
+    }
+
+    protected static function getEchoFalseReturnType( string $name ): Type
```

Comment:
```suggestion
    protected static function getEchoFalseReturnType(string $name): Type
```

This is not WordPress :)